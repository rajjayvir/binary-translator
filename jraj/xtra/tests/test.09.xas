  jmp start
display:
  push r0
  push r1
  loadi 0, r0
  loadi 0, r1
  std
  cld
  pop r1
  pop r0
  ret
start:
  loadi 0x1234, r0
  push r0
  mov r15, r0
  loadi 1, r1
  add r0, r1

  loadb r0, r2
  loadb r1, r3
  call display

  storb r2, r1
  storb r3, r0
  loadb r0, r2
  loadb r1, r3
  call display
  pop r0
.literal 0

