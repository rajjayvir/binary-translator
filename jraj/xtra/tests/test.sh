#!/bin/sh

if [ ! -x xas ]; then
  make xas 
fi

echo ======================================================
echo ====================== TEST $1 ========================
echo ======================================================
./xas tests/test.$1.xas tests/test.$1.xo > /dev/null
if timeout 10 ./$2/$3 tests/test.$1.xo > tests/test.$1.s; then 
   echo Translated tests/test.$1.s 
elif [ $? -eq 124 ]; then
  echo TIMEOUT
  exit 1
else 
  echo Abnormal program termination: the program crashed
  echo Exit code $?
  exit 1
fi

echo Compiling runner
gcc -o runner runner.c regsdump.s tests/test.$1.s
if [ ! -x runner ]; then
  echo runner not created.
  exit 1
fi

echo Running runner
if timeout 10 ./runner > tests/test.$1.out; then 
  if diff -w tests/test.$1.out tests/test.$1.expected > /dev/null; then
    echo PASSED
  else
    echo FAILED
    echo ======
    echo ____Your_output____ __Expected_Output___
   #echo +++++++++++++++++++ ++++++++++++++++++++
    diff -y -W 80 tests/test.$1.out tests/test.$1.expected 
    echo =====================
    echo Your Output:
    echo ++++++++++++++++
    cat tests/test.$1.out
    echo =====================
    echo Expected Output:
    echo ++++++++++++++++
    cat tests/test.$1.expected
    exit 1
  fi
elif [ $? -eq 124 ]; then
  echo TIMEOUT
  exit 1
else 
  echo Abnormal program termination: runner crashed
  echo Exit code $?
  exit 1
fi
