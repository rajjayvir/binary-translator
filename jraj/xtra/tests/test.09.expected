Registers:
  %rax: 0x0000000000000000    %rbx: 0x0000000000000000    
  %rcx: 0x0000000000000034    %rdx: 0x0000000000000012    
  %rsi: 0x0000000000000000    %rdi: 0x0000000000000000    
  %r8 : 0x0000000000000000    %r9 : 0x0000000000000000    
  %r10: 0x0000000000000000    %r11: 0x0000000000000000    
  %r12: 0x0000000000000000    %r13: 0x0000000000000000    
  %r14: 0x0000000000000000    %r15: 0x0000000000000000    
Registers:
  %rax: 0x0000000000000000    %rbx: 0x0000000000000000    
  %rcx: 0x0000000000000012    %rdx: 0x0000000000000034    
  %rsi: 0x0000000000000000    %rdi: 0x0000000000000000    
  %r8 : 0x0000000000000000    %r9 : 0x0000000000000000    
  %r10: 0x0000000000000000    %r11: 0x0000000000000000    
  %r12: 0x0000000000000000    %r13: 0x0000000000000000    
  %r14: 0x0000000000000000    %r15: 0x0000000000000000    
